tests/functional/test_charm.py

```diff
@@ -29,6 +30,40 @@ def get_this_script_dir() -> Path:
     return Path(path)
 
 
+def parse_metrics(metrics_input: str, collector_name: Optional[str] = None) -> list[str]:
+    """Parse raw metrics and return list of collector-specific metrics.
+
+    For example, if the metrics_input is
+
+    # ipmi_dcmi_metric1 Some comment about this
+    ipmi_dcmi_metric1 1.0
+    # ipmi_sel_metric2 Another comment
+    ipmi_sel_metric2 1.0
+    # ipmi_sel_metric3 Last comment
+    ipmi_sel_metric3 0
+
+    then parse_metrics(metrics_input, collector_name='ipmi_sel') would return
+    ['ipmi_sel_metric2 1.0', 'ipmi_sel_metric3 0']
+
+    All comments are removed and each metric along with its value are
+    stored as items in the list.
+    If collector_name is not provided, returns all metrics in a list.
+    """
+    parsed_metrics = []
+    if collector_name:
+        for metric in metrics_input.split("\n"):
+            if metric.startswith(collector_name):
+                parsed_metrics.append(metric)
```

Comment:
Nitpick:
What if we store this as a tuple like:
```suggestion
                parsed_metrics.append(tuple(metric.strip().split()))
```
I know that currently the string works, but if we want to validate the metric value (for example, verifying it's a number in a certain range), having it directly readable from a tuple is simpler than parsing from the string.

Reply:
Great suggestion! I'm sure as part of the tests, we'll need to be picking apart not just the metric value but the labels as well. To that extent, I'm in the process of creating a `Metric` dataclass with `name`, `value` and an optional labels. 

For example, `megaraid_physical_drive_info{controller_id="0",drive_type="SSD",enclosure_id="68",slot_id="0",state="Onln"} 1.0` would give me 
```
Metric(name='megaraid_physical_drive_info', labels='controller_id="0",drive_type="SSD",enclosure_id="68",slot_id="0",state="Onln"', value=1.0)
```

If labels is not provided, it will be provided as `None`.